### PR TITLE
add option for s3 bucket #minor

### PR DESCRIPTION
**Pull request by blankdots**
### Describe the pull request:

- [x] Functional change
- [x] New feature
- [x] Code cleanup

### Changes made:

- s3 now supports buckets
- enable custom path for POSIX FS in case another Filesystem is mounted as POSIX e.g. https://github.com/ctrox/csi-s3/ for k8s
- fix running tests
- other small fixes and formatting

### CODE REVIEW

File: src/main/java/se/nbis/lega/inbox/sftp/InboxSftpEventListener.java [hunk 1]
```diff
@@ -258,6 +259,10 @@ protected String getFilePath(Path path, String username) {
         }
         log.debug("POSIX filepath is {} for user {}", key, username);
 
+//        if (inboxFSPath != null && inboxFSPath != "") {
+//            return inboxFSPath + "/" + key    ;
+//        }
+
```
Comment (jbygdell) [inline src/main/java/se/nbis/lega/inbox/sftp/InboxSftpEventListener.java]: Commented out lines should be removed.

**Review by jbygdell — DISMISSED** [review]

File: src/main/java/se/nbis/lega/inbox/sftp/InboxSftpEventListener.java (hunk 1 above)
Comment (blankdots) [reply src/main/java/se/nbis/lega/inbox/sftp/InboxSftpEventListener.java]: actually that should have been there my mistake, now it is fixed

**Review by jbygdell — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -46,31 +46,33 @@ while S3 is the real final destination for the uploaded files.
...
+| KEYSTORE_PATH       | /etc/ega/inbox.jks | Path to Keystore file                                           |
+| KEYSTORE_PASSWORD   |                    | Password to access the Keystore                                 |
+| CACHE_TTL           | 300.0              | CEGA credentials time-to-live                                   |
+| CEGA_ENDPOINT       |                    | CEGA REST endpoint                                              |
+| CEGA_ENDPOINT_CREDS |                    | CEGA REST credentials                                           |
+| S3_ENDPOINT         | inbox-backend:9000 | Inbox S3 backend URL                                            |
+| S3_REGION           | us-east-1          | Inbox S3 backend region (us-east-1 is default in Minio)         |
+| S3_ACCESS_KEY       |                    | Inbox S3 backend access key (S3 disabled if not specified)      |
+| S3_SECRET_KEY       |                    | Inbox S3 backend secret key (S3 disabled if not specified)      |
+| S3_BUCKET           |                    | Inbox S3 backend secret keyucket (S3 disabled if not specified) |
```
Comment (pontus) [inline README.md]:
```suggestion
| S3_BUCKET           |                    | Inbox S3 backend secret bucket (S3 disabled if not specified) |
```

File: src/test/resources/s3-storage.application.properties [hunk 3]
```diff
@@ -17,9 +17,10 @@ inbox.s3.endpoint=${S3_ENDPOINT:localhost:9000}
 inbox.s3.region=${S3_REGION:us-west-2}
 inbox.s3.access-key=${S3_ACCESS_KEY:minio}
 inbox.s3.secret-key=${S3_SECRET_KEY:miniostorage}
+inbox.s3.bucket=${S3_SECRET_KEY:default}
```
Comment (pontus) [inline src/test/resources/s3-storage.application.properties]:
```suggestion
inbox.s3.bucket=${S3_BUCKET:default}
```

**Review by pontus — APPROVED** [review]

**Pull request merged**